internal/internal_event_handlers.go

```diff
@@ -203,6 +216,23 @@ func (wc *workflowEnvironmentImpl) RequestCancelWorkflow(domainName, workflowID,
 	return nil
 }
 
+func (wc *workflowEnvironmentImpl) SignalExternalWorkflow(domainName, workflowID, runID, signalName string,
+	input []byte, callback resultHandler) {
+
+	if domainName == "" {
+		callback(nil, errors.New("domain is empty"))
+		return
+	}
+	if workflowID == "" {
```

Comment:
do input validation in the workflow.SignalExternalWorkflow() method.

Reply:
changed